File: Go/6_Zigzag Conversion.go
import "fmt"

func convert(s string, numRows int) string {
	if numRows == 1 {
		return s
	}
	x := make([]string, numRows)
	cnt := 0
	flg := false
	for i := 0; i < len(s); i++ {
		//fmt.Println(cnt," ",i," ",string(s[i]),"  ",flg)
		x[cnt] = x[cnt] + string(s[i])
		if flg == false {
			cnt += 1
		} else {
			cnt -= 1
		}
		if cnt >= numRows {
			flg = true
			cnt -= 2
		}
		if cnt == 0 {
			flg = false
		}
	}
	result := ""
	for i := 0; i < len(x); i++ {
		result += x[i]
	}
	return result
}

File: Python/6_Zigzag_Conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows==1:
            return s
        res=[""]*numRows
        i=0
        flag=True
        for ch in s:
            if flag:
                if i<numRows-1:
                    res[i]+=ch
                    i+=1
                elif i==numRows-1:
                    res[i]+=ch
                    i-=1
                    flag=False
            else:
                if i>0:
                    res[i]+=ch
                    i-=1
                elif i==0:
                    res[i]+=ch
                    i+=1
                    flag=True
        return "".join(res)